src/Dsn.php

```diff
@@ -239,6 +239,18 @@ protected function parseUrlArray($url)
 
         $url = array_merge($this->uriKeys, $url);
 
+        if (isset($url['host'])) {
```

Comment:
Can you chnge this pleas to be 

    foreach['host', 'user', 'pass'] as $key) {
        if (!isset($url[$key]) {
            continue;
        }
        $url[$key] = urldecode($url[$key]);
    }

please - I.e. make is so that adding another key (should that be necessary in the future) is just adding to an array rather than copy and paste.